### DISCUSSION TITLE
[firebase-analytics] number parameters not sent

### DISCUSSION THREAD

**Discussion by louis030195** [firebase/firebase-js-sdk#6433] [Q&A] [answered]
Hi, I have been using Google/Firebase Analytics for some time and properly receiving my events with the string values, but when I try to use "number" values it's always kind of bugged, 0 or null valued.

```ts
const analytics = useAnalytics();
logEvent(analytics, "drawer_tab", {
  current_drawer_tab: drawerSelection,
  new_drawer_tab: value,
  role: role,
  product_id: productData?.id || "",
  elapsed_time: drawerTabTimer.elapsedTime, // Here elapsedTime is always a float
});
console.log("drawerTabTimer.elapsedTime", drawerTabTimer.elapsedTime); // always shows correct float value
```

In Google Analytics, I can't ever see values, even though I have some string values in BigQuery afterward.

<img width="1060" alt="Screenshot 2022-07-13 at 10 32 45" src="https://user-images.githubusercontent.com/25003283/178688607-2185afab-7c56-4feb-be83-53bd51897399.png">

I use the Google Cloud BigQuery integration

```sql
SELECT 
  event_date,
  event_timestamp,
  event_params
FROM `foo.bar.events_intraday_*`
WHERE 
_table_suffix BETWEEN "20220700" AND "20220715"
ORDER BY event_timestamp DESC
```

<img width="1118" alt="Screenshot 2022-07-13 at 10 30 39" src="https://user-images.githubusercontent.com/25003283/178688180-e4689351-067c-4794-a4db-f2f1dc8a9ac4.png">

Is there an issue with Analytics not dealing well with number/float or something like that? Or should I even cast numbers to string?

Thanks a lot!

**Comment by hsubox76** [comment]
The Firebase Analytics web SDK is a wrapper around gtag.js that passes the params directly to gtag with no additional formatting, and according to the gtag documentation, numbers are fine. https://developers.google.com/analytics/devguides/collection/gtagjs/events

It looks like from the documentation that if you want it to show up as the "event value" in the analytics dashboard you have to make the key literally `value`. Were you doing that before and just changed the key names for this bug report? As for it being null in BigQuery results, I'm not familiar with BigQuery so I don't know if something has gone wrong in the pipeline there.

**Comment by louis030195** [comment, marked as answer]
In the end, I just waited a few more days and I got the double value now
